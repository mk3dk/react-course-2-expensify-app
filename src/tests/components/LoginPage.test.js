import React from 'react';
import {shallow} from 'enzyme';
import {LoginPage} from '../../components/LoginPage';
import {Header} from "./Header.test";

let startLogin, wrapper;

beforeEach(() => {
    startLogin = jest.fn();
    wrapper = shallow(<LoginPage startLogin={startLogin}/>);
});

test('should correctly render LoginPage', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
    wrapper.find('button').simulate('click');;
    expect(startLogin).toHaveBeenCalled();
});
