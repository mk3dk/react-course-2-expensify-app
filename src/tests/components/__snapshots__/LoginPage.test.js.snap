// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render LoginPage 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;

exports[`should render Header correctly 1`] = `undefined`;
